### PR TITLE
chore: add pygoat tests

### PR DESCRIPTION
**Pull request by juanjux**
## Description

This PR adds integration testing of Code Vulnerability features using pygoat. 

## Checklist

- [X] Change(s) are motivated and described in the PR description
- [X] Testing strategy is described if automated tests are not included in the PR
- [X] Risks are described (performance impact, potential for breakage, maintainability)
- [X] Change is maintainable (easy to change, telemetry, documentation)
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [X] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/appsec/appsec/pygoat_tests/test_pygoat.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    csrftoken = client.cookies["csrftoken"]
+    return client, csrftoken
+
+
+def get_traces(agent_client: requests.Session) -> requests.Response:
+    return agent_client.get(TESTAGENT_URL + "/traces" + TESTAGENT_TOKEN_PARAM, headers=TESTAGENT_HEADERS)
+
+
+def vulnerability_in_traces(vuln_type: str, agent_client: requests.Session) -> bool:
+    time.sleep(5)
```
Comment (gnufede) [inline tests/appsec/appsec/pygoat_tests/test_pygoat.py:75]: isn't this a lot sleeping?

**Review by gnufede — APPROVED** [review]

File: tests/appsec/appsec/pygoat_tests/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+version: "3"
```
Comment (gnufede) [inline tests/appsec/appsec/pygoat_tests/docker-compose.yml:1]: Can't we use the global docker-compose? Like
https://github.com/DataDog/dd-trace-py/blob/main/docker-compose.yml

Comment (juanjux) [reply tests/appsec/appsec/pygoat_tests/docker-compose.yml]: Done!

**Review by avara1986 — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -181,6 +181,21 @@ services:
...
+        ports:
+            - "127.0.0.1:8000:8000"
+        environment:
+            - DD_APPSEC_ENABLED=true
+            - DD_IAST_ENABLED=true
+            - DD_IAST_REQUEST_SAMPLING=100
+            - DD_IAST_VULNERABILITIES_PER_REQUEST=100
+            - DD_REMOTE_CONFIGURATION_ENABLED=true
+            - DD_AGENT_PORT=8126
+            - DD_TRACE_AGENT_URL=http://testagent:8126
```
Comment (romainkomorndatadog) [inline docker-compose.yml:197]: Are `DD_AGENT_PORT` and `DD_TRACE_AGENT` both necessary?

File: tests/appsec/integrations/pygoat_tests/Dockerfile.pygoat.2.0.1 [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+COPY . /app/
+
+# copy and install ddtrace
+WORKDIR /
+COPY ../../../../ ddtrace
+RUN pip install --no-cache-dir ddtrace
+
+WORKDIR /app/pygoat
+# Install dependencies
+RUN git checkout v2.0.1
```
Comment (romainkomorndatadog) [inline tests/appsec/integrations/pygoat_tests/Dockerfile.pygoat.2.0.1:31]: nit: I'd move this up to the "download" section (or move the cloning down to here).

File: docker-compose.yml (hunk 3 above)
Comment (juanjux) [reply docker-compose.yml]: Seem like its not, thanks!

**Review by brettlangdon — APPROVED** [review]
docker compose changes are 👍🏻

**Pull request merged**